libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1032,6 +1032,62 @@ string YulUtilFunctions::suffixedVariableNameList(string const& _baseName, size_
 	return result;
 }
 
+std::string YulUtilFunctions::decrementCheckedFunction(Type const& _type)
+{
+	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+
+	string const functionName = "decrement_" + _type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		u256 minintval;
+
+		// Smallest admissible value to decrement
+		if (type.isSigned())
+			minintval = 0 - (u256(1) << (type.numBits() - 1)) + 1;
```

Comment:
I'm still a bit worried about the compiler error message that was produced when you still had `-(u256...` - is it really undefined behaviour for u265?

Reply:
Not sure what to say.. at least it didn't warn they way it is written now?

Reply:
It seems to be fine:

> Unsigned types support subtraction - the result is "as if" a 2's complement operation had been performed as long as they are not checked-integers (see above). In other words they behave pretty much as a built in integer type would in this situation. So for example if we were using uint128_t then uint128_t(1)-4 would result in the value 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFD of type uint128_t. However, had this operation been performed on checked_uint128_t then a std::range_error would have been thrown.
> Unary negation of unsigned types results in a compiler error (static assertion). 

(from https://www.boost.org/doc/libs/1_70_0/libs/multiprecision/doc/html/boost_multiprecision/tut/ints/cpp_int.html )